public/src/admin.js

```diff
@@ -0,0 +1,28 @@
+const settings = require('settings');
+const hooks = require('hooks');
+
+hooks.onPage('action:ajaxify.end', () => {
+	// note: this is probably deprecated in core soon...
+	socket.emit('categories.get', function(err, data) {
+		categories = data;
+		for (var i = 0; i < categories.length; ++i) {
+			$('#postCategories').append('<option value=' + categories[i].cid + '>' + categories[i].name + '</option>');
+		}
+	});
+})
+
+settings.load('discord-notification', $('.discord-notification-settings'));
+
+$('#save').on('click', function() {
+	settings.save('discord-notification', $('.discord-notification-settings'), function() {
+		app.alert({
+			type: 'success',
+			alert_id: 'discord-notification-saved',
+			title: 'Settings Saved',
+			message: 'Please reload your NodeBB to apply these settings',
+			clickfn: function() {
+				socket.emit('admin.reload');
+			}
+		});
+	});
+});
```

Comment:
Everything in this PR is compatible with NodeBB v1.x **except for this file**.

But if you want to you can convert it back to AMD format by adding the necessary `define()` wrapper.

Reply:
So if you do that, you can change `nbbpm.compatibility` back to what it was before.